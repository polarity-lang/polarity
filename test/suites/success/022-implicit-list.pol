/// The type of finite lists.
data List(implicit a: Type) {
    /// The canonical empty list.
    Nil(implicit a: Type): List(a:=a),
    /// Appending one element to the front of a list.
    Cons(implicit a: Type, x: a, xs: List(a:=a)): List(a:=a)
}

data Top { Unit }

let example: List() {Cons(Unit, Nil())}
