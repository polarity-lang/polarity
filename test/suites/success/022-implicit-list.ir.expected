let example { Cons(<ZST>, Unit, Nil(<ZST>)) }
